adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -263,7 +269,19 @@ else if (response.containsKey(AuthenticationConstants.OAuth2.ACCESS_TOKEN)) {
                     response.get(AuthenticationConstants.OAuth2.ACCESS_TOKEN),
                     response.get(AuthenticationConstants.OAuth2.REFRESH_TOKEN), expires.getTime(),
                     isMultiResourceToken, userinfo, tenantId, rawIdToken);
-            
+
+            if (response.containsKey(AuthenticationConstants.OAuth2.EXT_EXPIRES_IN)) {
+                result.setIsExtendedLifeTimeToken(true);
```

Comment:
Shouldn't you only set this flag when you're returning the stale token to developer?